src/components/Select/SelectBase/SelectBase.tsx

```diff
@@ -0,0 +1,268 @@
+    }) => ReactNode;
+}
+
+const { block, elem } = bem('SelectBase', styles);
+
+export function SelectBase<S>(props: Props<S>) {
+    const {
+        suggestions,
+        suggestionToString,
+        clearTitle,
+        showClearButton,
+        noSuggestionsPlaceholder,
+        onBlur,
+        onSelectionChange,
+        onInputValueChange,
+        onClearAllSelected,
+        inputRef: inputRefFromProps,
+        rootRef: rootRefFromProps,
+        listRef: listRefFromProps,
+        listRenderer,
+        focusedRenderer,
+        blurredRenderer,
+        keepExpandedAfterSelection,
+        ...rest
+    } = props;
+
+    const [inputRef, setInputRef] = React.useState(
+        inputRefFromProps || React.createRef<HTMLInputElement>()
+    );
+    const [rootRef, setRootRef] = React.useState(
+        rootRefFromProps || React.createRef<HTMLDivElement>()
+    );
+    const [listRef, setListRef] = React.useState(
+        listRefFromProps || React.createRef<HTMLUListElement>()
+    );
+
+    const [inputValue, setInputValue] = React.useState('');
+    const [inputValueRecall, setInputValueRecall] = React.useState('');
+    const [focused, setFocused] = React.useState(false);
+
+    // focus input field if component is focused
+    React.useEffect(() => {
+        if (focused) {
+            inputRef.current?.focus();
+        }
+    }, [focused, inputRef]);
+
+    // update inputRef from props if changed
+    React.useEffect(() => {
+        if (inputRefFromProps) {
+            setInputRef(inputRefFromProps);
+        }
+    }, [inputRefFromProps]);
+
+    // update listRef from props if changed
+    React.useEffect(() => {
+        if (listRefFromProps) {
+            setListRef(listRefFromProps);
+        }
+    }, [listRefFromProps]);
+
+    // update rootRef from props if changed
+    React.useEffect(() => {
+        if (rootRefFromProps) {
+            setRootRef(rootRefFromProps);
+        }
+    }, [rootRefFromProps]);
+
+    const handleBlur = () => {
+        setInputValue('');
+        setInputValueRecall('');
+        onBlur?.();
+    };
+
+    const handleChange = selectedItem => {
+        setInputValue('');
+
+        if (selectedItem) {
+            onSelectionChange(selectedItem);
+        }
+
+        if (!keepExpandedAfterSelection) {
+            handleBlur();
+        }
+    };
+
+    const handleInputValueChange = (inputVal, changes) => {
+        if (changes.type === Downshift.stateChangeTypes.changeInput) {
+            setInputValue(inputVal);
+            setInputValueRecall(inputVal);
+            if (listRef.current) {
+                listRef.current.scrollTop = 0;
+            }
+
+            onInputValueChange?.(inputVal);
+        }
+    };
+
+    const handleClearSelectedSuggestions = e => {
+        e.stopPropagation();
+        onClearAllSelected?.();
+    };
+
+    const focus = openMenu => {
+        openMenu();
+        inputRef.current?.focus();
```

Comment:
maybe need to set the focused state as well. If you do that, there is no need to set the input in focus here, the useEffect will take care of that